src/librustc/ty/maps/on_disk_cache.rs

```diff
@@ -223,6 +231,45 @@ impl<'sess> OnDiskCache<'sess> {
                 encode_query_results::<trans_fn_attrs, _>(tcx, enc, qri)?;
             }
 
+            // encode successful constant evaluations
+            let constants_index = {
```

Comment:
It would be great if we had a comment (here or elsewhere) explaining why constants even need special handling (like you did on IRC).